diff --git a/more_itertools/more.py b/more_itertools/more.py
index 737b07d..6fc8019 100755
--- a/more_itertools/more.py
+++ b/more_itertools/more.py
@@ -3129,20 +3129,7 @@ def exactly_n(iterable, n, predicate=bool):
     so avoid calling it on infinite iterables.
 
     """
-    iterator = filter(predicate, iterable)
-    if n <= 0:
-        if n < 0:
-            return False
-        for _ in iterator:
-            return False
-        return True
-
-    iterator = islice(iterator, n - 1, None)
-    for _ in iterator:
-        for _ in iterator:
-            return False
-        return True
-    return False
+    return ilen(islice(filter(predicate, iterable), n + 1)) == n
 
 
 def circular_shifts(iterable, steps=1):
