diff --git a/more_itertools/more.py b/more_itertools/more.py
index 737b07d..5ef5173 100755
--- a/more_itertools/more.py
+++ b/more_itertools/more.py
@@ -553,10 +553,9 @@ def iterate(func, start):
     [10, 5, 16, 8, 4, 2, 1]
 
     """
-    with suppress(StopIteration):
-        while True:
-            yield start
-            start = func(start)
+    while True:
+        yield start
+        start = func(start)
 
 
 def with_iter(context_manager):
