diff --git a/more_itertools/more.py b/more_itertools/more.py
index 737b07d..eebffb2 100755
--- a/more_itertools/more.py
+++ b/more_itertools/more.py
@@ -286,7 +286,7 @@ def last(iterable, default=_marker):
         if isinstance(iterable, Sequence):
             return iterable[-1]
         # Work around https://bugs.python.org/issue38525
-        if getattr(iterable, '__reversed__', None):
+        if hasattr(iterable, '__reversed__'):
             return next(reversed(iterable))
         return deque(iterable, maxlen=1)[-1]
     except (IndexError, TypeError, StopIteration):
