diff --git a/more_itertools/more.py b/more_itertools/more.py
index 737b07d..3e06ae4 100755
--- a/more_itertools/more.py
+++ b/more_itertools/more.py
@@ -4380,7 +4380,7 @@ def product_index(element, *iterables, repeat=1):
     """
     elements = tuple(element)
     pools = tuple(map(tuple, iterables)) * repeat
-    if len(elements) != len(pools):
+    if len(element) != len(pools):
         raise ValueError('element is not a product of args')
 
     index = 0
